substratevm/mx.substratevm/mx_substratevm.py

```diff
@@ -1877,6 +1890,35 @@ def checkLine(line, marker, init_kind, msg, wrongly_initialized_lines):
 
     native_image_context_run(build_and_test_clinittest_image, args)
 
+class NoJlinkModuleInfoCompilationParticipant:
+
+    def __init__(self, dist, module_name):
+        self.dist = dist
+        self.module_name = module_name
+
+    # Upgrade module path for compilation of module-info.java files when not using jmods from the JDK
+    def __process__(self, module_desc):
+        """
+        :param module_desc: The JavaModuleDescriptor for this distribution
+        :rtype: list of strings with extra javac arguments
+        """
+        has_module_dep = False
+        graal_mod = None
+        for m in module_desc.modulepath:
+            if m.name == self.module_name:
+                has_module_dep = True
+                graal_mod = m
+                break
+        if has_module_dep:
+            return [ '--upgrade-module-path=' + self.safe_path_arg(graal_mod.jarpath) ]
```

Comment:
Once you did https://github.com/oracle/graal/pull/10161/files#r1862209294
```suggestion
            return [ '--upgrade-module-path=' + safe_path_arg(graal_mod.jarpath) ]
```